runtime/Java/src/org/antlr/v4/runtime/UnbufferedCharStream.java

```diff
@@ -82,7 +82,7 @@ public UnbufferedCharStream() {
 	/** Useful for subclasses that pull char from other than this.input. */
 	public UnbufferedCharStream(int bufferSize) {
 		n = 0;
-		data = new char[bufferSize];
+		data = new int[bufferSize];
 	}
 
 	public UnbufferedCharStream(InputStream input) {
```

Comment:
Yes! We absolutely should add a `Charset` arg, or there's no way to use anything but the default in the calling environment. (We should probably also change to default to UTF-8..)

`fill()` uses the `InputStreamReader` to get UTF-16 encoded `char`s. It will work with any encoding, although there's no way to specify an encoding other than the environment default with the current `UnbufferedCharStream` API.